components/datetime/src/raw/neo.rs

```diff
@@ -241,16 +241,39 @@ impl TimePatternSelectionData {
         length: MaybeLength,
         components: NeoTimeComponents,
     ) -> Result<Self, DataError> {
-        let payload = provider
-            .load_bound(DataRequest {
+        // First try to load with the explicit hour cycle. If there is no explicit hour cycle,
+        // or if loading the explicit hour cycle fails, then load with the default hour cycle.
+        let mut maybe_payload = None;
+        if let Some(hc) = locale.get_unicode_ext(&icu_locale_core::extensions::unicode::key!("hc"))
```

Comment:
suggestion: resolve the `hc` at the top-level constructor, and pass it separately from `DataLocale`. This will make it easier to remove Unicode extensions from `DataLocale` and migrate the input `DataLocale` to preferences.

Reply:
Done